packages/api/src/WithIntegrationHelpers.tsx

```diff
@@ -0,0 +1,352 @@
+      : Promise.resolve(integration);
+  }
+
+  public setName(integration: Integration, name: string): Integration {
+    return produce(integration, nextIntegration => {
+      nextIntegration.name = name;
+    });
+  }
+
+  public makeLocalStorageId(id: string) {
+    return `iec-${id}`;
+  }
+
+  public async createDraft(value: Integration) {
+    // TODO: this should be handled by the BE
+    if (!value.id) {
+      throw new Error("Integration has no id, can't create a draft");
+    }
+    const id = value.id;
+    await this.setDraft(id, value);
+    return Promise.resolve(id);
+  }
+
+  public getDraft(id: string): Promise<Integration> {
+    // TODO: this should be handled by the BE
+    const serializedIntegration = localStorage.getItem(
+      this.makeLocalStorageId(id)
+    );
+    if (!serializedIntegration) {
+      throw new Error(`There is no draft for id ${id}`);
+    }
+    return Promise.resolve(deserializeIntegration(serializedIntegration));
+  }
+
+  public setDraft(id: string, value: Integration): Promise<void> {
+    // TODO: this should be handled by the BE
+    localStorage.setItem(
+      this.makeLocalStorageId(id),
+      serializeIntegration(value)
+    );
+    return Promise.resolve();
+  }
+
+  public getSteps(value: Integration, flow: number): Step[] {
+    try {
+      return value.flows![flow].steps!;
+    } catch (e) {
+      throw new Error(`Can't find steps in position flow:${flow}`);
+    }
+  }
+
+  public getStep(value: Integration, flow: number, step: number): Step {
+    try {
+      return value.flows![flow].steps![step];
+    } catch (e) {
+      throw new Error(
+        `Can't find a step in position flow:${flow} step:${step}`
+      );
+    }
+  }
+
+  public canPublish(integration: IntegrationOverview) {
+    return integration.currentState !== 'Pending';
+  }
+
+  public canActivate(integration: IntegrationOverview) {
+    return (
+      integration.currentState !== 'Pending' &&
+      integration.currentState !== 'Published'
+    );
+  }
+
+  public canEdit(integration: IntegrationOverview) {
+    return integration.currentState !== 'Pending';
+  }
+
+  public canDeactivate(integration: IntegrationOverview) {
+    return integration.currentState !== 'Unpublished';
+  }
+
+  public render() {
+    return this.props.children({
+      addConnection: this.addConnection,
+      canActivate: this.canActivate,
+      canDeactivate: this.canDeactivate,
+      canEdit: this.canEdit,
+      canPublish: this.canPublish,
+      createDraft: this.createDraft,
+      getDraft: this.getDraft,
+      getEmptyIntegration: this.getEmptyIntegration,
+      getStep: this.getStep,
+      getSteps: this.getSteps,
+      saveIntegration: this.saveIntegration,
+      setDraft: this.setDraft,
+      setName: this.setName,
+      updateConnection: this.updateConnection,
+      updateOrAddConnection: this.updateOrAddConnection,
+    });
+  }
+}
+
+export const WithIntegrationHelpers: React.FunctionComponent<
```

Comment:
Just a general question (I think we actually talked about this), where/how are all the errors thrown within this component handled?